rero_ils/modules/selfcheck/utils.py

```diff
@@ -62,3 +65,91 @@ def format_patron_address(patron):
         postal_code=patron.get('postal_code'),
         city=patron.get('city')
     )
+
+
+def get_patron_status(patron):
+    """Return patron status useful for sip2.
+
+    * check if the user is blocked ?
+    * check if the user reaches the maximum loans limit ?
+    * check if the user reaches the maximum fee amount limit ?
+
+    add PatronStatusType e.g.:
+        patron_status.add_patron_status_type(
+            SelfcheckPatronStatusTypes.CARD_REPORTED_LOST)
+
+    :return SelfcheckPatronStatus object or None.
+    """
+    if check_sip2_module():
+        from invenio_sip2.models import SelfcheckPatronStatus
+
+        patron_status = SelfcheckPatronStatus()
+        # check if patron is blocked
+        if patron.is_blocked:
+            patron_status.add_patron_status_type(
+                SelfcheckPatronStatusTypes.CHARGE_PRIVILEGES_DENIED)
+            patron_status.add_patron_status_type(
+                SelfcheckPatronStatusTypes.RENEWAL_PRIVILEGES_DENIED)
+            patron_status.add_patron_status_type(
+                SelfcheckPatronStatusTypes.RECALL_PRIVILEGES_DENIED)
+            patron_status.add_patron_status_type(
+                SelfcheckPatronStatusTypes.HOLD_PRIVILEGES_DENIED)
+
+        patron_type = PatronType.get_record_by_pid(patron.patron_type_pid)
+        # check the patron type checkout limit
+        if not patron_type.check_checkout_count_limit(patron):
+            patron_status.add_patron_status_type(
+                SelfcheckPatronStatusTypes.TOO_MANY_ITEMS_CHARGED)
+        # check the patron type fee amount limit
+        if not patron_type.check_fee_amount_limit(patron):
+            patron_status.add_patron_status_type(
+                SelfcheckPatronStatusTypes.EXCESSIVE_OUTSTANDING_FINES)
+            patron_status.add_patron_status_type(
+                SelfcheckPatronStatusTypes.EXCESSIVE_OUTSTANDING_FEES)
+        # check the patron type overdue limit
+        if not patron_type.check_overdue_items_limit(patron):
+            patron_status.add_patron_status_type(
+                SelfcheckPatronStatusTypes.TOO_MANY_ITEMS_OVERDUE)
+
+        return patron_status
+
+
+def map_media_type(media_type):
+    """Get mapped media type.
+
+    :param media_type: Document type
+    :return: sip2 media type (see invenio_sip2.models.SelfcheckMediaType)
+    """
+    media_types = {
+         'article': 'MAGAZINE',
+         'book': 'BOOK',
+         'journal': 'MAGAZINE',
+         'sound': 'AUDIO',
+         'video': 'VIDEO',
+    }
```

Comment:
In my sense, In next PR, it will be smart to place this mapping into the `config.py` file

Reply:
ok